src/peer_manager.rs

```diff
@@ -1073,7 +1073,17 @@ impl PeerManager {
 
     /// Removes the given peer. Returns whether the peer was actually present.
     pub fn remove_peer(&mut self, pub_id: &PublicId) -> bool {
-        self.peers.remove(pub_id).is_some()
+        if let Some(mut peer) = self.peers.remove(pub_id) {
+            if peer.is_joining_node() && peer.is_node() {
```

Comment:
...then use the function here: `if peer.was_joining_node { ...`  (or whatever you decide to call it).